plaidml/bridge/openvino/ops/roi_align.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+
+#include "ngraph/opsets/opset3.hpp"
+#include "plaidml/op/op.h"
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace {
+
+template <typename T>
+std::vector<T> cast_constant_operand(size_t operand_idx, ngraph::Node* layer) {
+  auto* ngraph_const = ngraph::as_type<ngraph::op::Constant>(layer->get_input_node_ptr(operand_idx));
+  if (ngraph_const) {
+    return ngraph_const->cast_vector<T>();
+  } else {
+    THROW_IE_EXCEPTION
+        << "Dynamic slicing not currently supported by PlaidML plugin; all of indices, offsets and default index"
+           "must be Constants.";
+  }
+}
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerROIAlign() {
+  registerOp("roialign", [](const Context& ctx) {
+    IE_ASSERT(ctx.operands.size() == 3);
+    auto X = ctx.operands.at(0);
+    auto* layer = ngraph::as_type<ngraph::opset3::ROIAlign>(ctx.layer);
+    auto boxes = cast_constant_operand<int32_t>(1, layer);
+    auto batch_indices = cast_constant_operand<int32_t>(2, layer);
+
+    auto sampling_ratio = layer->get_sampling_ratio();
+    IE_ASSERT(sampling_ratio == 0);
```

Comment:
As we only use `2*2` shape kernel to sample ROI and do pooling to output each feature map element, and `pooled_h` and `pooled_w` are given. The feature map before pooling layer and after interpolation on ROI should be in shape `2*pooled_h * 2*pooled_w`, `sampling_ratio` seems not needed here.